diff --git a/api/types.go b/api/types.go
index c0e771d..c8f4139 100644
--- a/api/types.go
+++ b/api/types.go
@@ -593,6 +593,7 @@ type Runner struct {
 	UseMMap         *bool `json:"use_mmap,omitempty"`
 	NumThread       int   `json:"num_thread,omitempty"`
 	DraftNumPredict int   `json:"draft_num_predict,omitempty"`
+	NumParallel     int   `json:"num_parallel,omitempty"`
 }
 
 // EmbedRequest is the request passed to [Client.Embed].
@@ -1117,6 +1118,7 @@ func DefaultOptions() Options {
 			NumGPU:          -1, // -1 here indicates that NumGPU should be set dynamically
 			NumThread:       0,  // let the runtime decide
 			DraftNumPredict: 4,
+			NumParallel:     int(envconfig.NumParallel()),
 			UseMMap:         nil,
 		},
 	}
